packages/components/src/internal/components/editable/EditableGridPanelForUpdate.tsx

```diff
@@ -31,21 +40,29 @@ type Models = {
 
 interface Props {
     containerFilter?: Query.ContainerFilter;
+    editStatusData?: OperationConfirmationData;
```

Comment:
nit: Consider altering this interface to actually extend `EditGridPanelProps`:

```tsx
type InheritedEditableGridPanelProps = Omit<
    EditableGridPanelProps,
    | 'columnMetadata'
    | 'editorModel'
    | 'forUpdate'
    | 'model'
    | 'onChange'
>;

interface EditableGridPanelForUpdateProps extends InheritedEditableGridPanelProps {
    editStatusData?: OperationConfirmationData;
    idField: string;
    loader: EditableGridLoader;
    onCancel: () => void;
    onComplete: () => void;
    pluralNoun?: string;
    queryModel: QueryModel;
    selectionData: Map<string, any>;
    singularNoun?: string;
    updateRows: (
        schemaQuery: SchemaQuery,
        rows: Array<Record<string, any>>,
        comment: string,
        originalRows: Array<Record<string, any>>
    ) => Promise<any>;
}
```

then the prop loadout within the component becomes:

```tsx
const { editStatusData, idField, loader, onCancel, onComplete, pluralNoun, queryModel, selectionData, singularNoun, updateRows, ...editableGridProps } = props;
const { getIsDirty, setIsDirty } = editableGridProps;
```

and finally the component is initialized with the proper overrides:

```tsx
<EditableGridPanel
    allowAdd={false}
    allowRemove={false}
    bordered
    bsStyle="info"
    striped
    title={`Edit selected ${pluralNoun}`}
    {...editableGridProps}
    columnMetadata={columnMetadata}
    editorModel={models.editorModel}
    forUpdate
    model={models.dataModel}
    onChange={onGridChange}
/>
```

Reply:
done